plugin-server/src/utils/db/db.ts

```diff
@@ -630,7 +664,15 @@ export class DB {
         return undefined
     }
 
-    private async getGroupProperty(teamId: number, groupIdentifier: GroupIdentifier): Promise<GroupProperties> {
+    private async getGroupCreatedAt(teamId: number, groupIdentifier: GroupIdentifier): Promise<GroupCreatedAt> {
```

Comment:
Meta: The db class is becoming a mess: It used to be pure functions which fetched values from db, now it's intermixed with business logic and caching.

There's competing methods for everything, making it really hard to grasp what method to actually use to achieve any goal.

Is there a way we could fix this? E.g. move the business-logic-like methods which deal with caches somewhere else or mixins or...?

Reply:
We could take a posthog-js esque approach (`posthog.people.set`) and move cache methods to a `Cache` class that's also a property of DB?

Reply:
After thinking about it a bit: I think we should extract createEvent from process-event.ts and couple all of the business logic/caching-related code in a single module along with it.

We can still keep some postgres/clickhouse fetch code in db.ts but caching is then localized to where we read/write it.

Thoughts? :)

Reply:
Makes some sense yes - one thing is that cache functions particularly are hit from other places. I think a Cache lib would also be useful as we could make sure Redis doesn't mix with the other DBs and we can get rid of all the lpush llen stuff from db as well

Reply:
I would caution against over-generalizing - caching logic in this case is going to be 100% coupled with the business logic here. If the caching logic is hit from elsewhere (e.g. person-state) then we can just update those callsites to refer to the module we'd be creating here.

Keep closely coupled things together.